wlog/auditlog/audit3log/params.go

```diff
@@ -0,0 +1,201 @@
+
+type paramStruct struct {
+	param auditloginternal.Audit3Param
+}
+
+func (p *paramStruct) getParam() auditloginternal.AuditParam {
+	return p.param.Param
+}
+
+func convertInternalParamToExportedParam(param auditloginternal.Audit3Param) Param {
+	return &paramStruct{
+		param: param,
+	}
+}
+
+func convertExternalParamsToInternalParams(params []Param) []auditloginternal.Audit3Param {
+	if params == nil {
+		return nil
+	}
+	out := make([]auditloginternal.Audit3Param, len(params))
+	for i := range out {
+		out[i] = auditloginternal.Audit3Param{
+			Param: params[i].getParam(),
+		}
+	}
+	return out
+}
+
+func Deployment(deployment string) Param {
+	return convertInternalParamToExportedParam(auditloginternal.Audit3Deployment(deployment))
+}
+
+func Host(host string) Param {
+	return convertInternalParamToExportedParam(auditloginternal.Audit3Host(host))
+}
+
+func Product(product string) Param {
+	return convertInternalParamToExportedParam(auditloginternal.Audit3Product(product))
+}
+
+func ProductVersion(productVersion string) Param {
+	return convertInternalParamToExportedParam(auditloginternal.Audit3ProductVersion(productVersion))
+}
+
+func Stack(stack string) Param {
+	return convertInternalParamToExportedParam(auditloginternal.Audit3Stack(stack))
+}
+
+func Service(service string) Param {
+	return convertInternalParamToExportedParam(auditloginternal.Audit3Service(service))
+}
+
+func Environment(environment string) Param {
+	return convertInternalParamToExportedParam(auditloginternal.Audit3Environment(environment))
+}
+
+func ProducerType(producerType AuditProducerType) Param {
+	return convertInternalParamToExportedParam(auditloginternal.Audit3ProducerType(auditloginternal.Audit3AuditProducerType(producerType)))
+}
+
+func Organizations(organizations []Organization) Param {
+	var audit3Organizations []auditloginternal.Audit3Organization
+	if len(organizations) > 0 {
+		audit3Organizations = make([]auditloginternal.Audit3Organization, len(organizations))
+		for i, org := range organizations {
+			audit3Organizations[i] = auditloginternal.Audit3Organization(org)
+		}
+	}
+	return convertInternalParamToExportedParam(auditloginternal.Audit3Organizations(audit3Organizations))
+}
+
+func EventID(eventID string) Param {
+	return convertInternalParamToExportedParam(auditloginternal.Audit3EventID(eventID))
+}
+
+func LogEntryID(logEntryID string) Param {
+	return convertInternalParamToExportedParam(auditloginternal.Audit3LogEntryID(logEntryID))
+}
+
+func UserAgent(userAgent string) Param {
+	return convertInternalParamToExportedParam(auditloginternal.Audit3UserAgent(userAgent))
+}
+
+// unexported because this should only be set by the "Category" param
+func categories(categories ...string) Param {
+	return convertInternalParamToExportedParam(auditloginternal.Audit3Categories(categories))
+}
+
+func Entities(entities []any) Param {
+	return convertInternalParamToExportedParam(auditloginternal.Audit3Entities(entities))
+}
```

Comment:
For entities, foundry-audit will extract rids from `Resource`s and only include the `rid`s in the audit log: https://github.palantir.build/foundry/foundry-audit/blob/develop/audit-core-api/src/main/java/com/palantir/foundry/audit/api/common/v2/EntityExtractor.java#L10 https://github.palantir.build/foundry/foundry-audit/blob/baf25b651addacd7fb83ccd43c1ad7a36cb27d0e/audit-core-api/src/main/java/com/palantir/foundry/audit/AuditLogEntryImpl.java#L182 Is it possible to do the same here

Reply:
Thanks, will take a look